Geometry_Engine/Modify/RoundCoordinates.cs

```diff
@@ -0,0 +1,323 @@
+        public static Polyline RoundCoordinates(this Polyline polyline, int decimalPlaces = 6)
+        {
+            return new Polyline()
+            {
+                ControlPoints = polyline.ControlPoints.Select(x => x.RoundCoordinates(decimalPlaces)).ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry PolyCurve's curves to be rounded to the number of provided decimal places.")]
+        [Input("polycurve", "The BHoM Geometry PolyCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry PolyCurve.")]
+        public static PolyCurve RoundCoordinates(this PolyCurve polycurve, int decimalPlaces = 6)
+        {
+            return new PolyCurve()
+            {
+                Curves = polycurve.Curves.Select(x => x.IRoundCoordinates(decimalPlaces)).ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry NurbsCurve's control points to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry NurbsCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry NurbsCurve.")]
+        public static NurbsCurve RoundCoordinates(this NurbsCurve nurbscurve, int decimalPlaces = 6)
+        {
+            return new NurbsCurve()
+            {
+                ControlPoints = nurbscurve.ControlPoints.Select(x => x.RoundCoordinates(decimalPlaces)).ToList(),
+                Knots = nurbscurve.Knots.ToList(),
+                Weights = nurbscurve.Weights.ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry PlanarSurface's defining curves to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry PlanarSurface to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("surface", "The modified BHoM Geometry PlanarSurface.")]
+        public static PlanarSurface RoundCoordinates(this PlanarSurface planarSurface, int decimalPlaces = 6)
+        {
+            Vector normal = planarSurface.Normal().Normalise();
+
+            if (Math.Abs(Math.Abs(normal.X) - 1) < Tolerance.Distance ||
+                Math.Abs(Math.Abs(normal.Y) - 1) < Tolerance.Distance ||
+                Math.Abs(Math.Abs(normal.Z) - 1) < Tolerance.Distance)
+            {
+                ICurve externalBoundery = planarSurface.ExternalBoundary.IRoundCoordinates(decimalPlaces);
+                List<ICurve> internalBounderies = planarSurface.InternalBoundaries.Select(x => x.IRoundCoordinates(decimalPlaces)).ToList();
```

Comment:
Control points of these curves may be rounded to two different values due to noise, e.g.:
- point A being (1.0051,0,0)
- point B being (1.0049,1,0)

They will have different X coordinates when rounded to 2 digits, resulting in a twisted surface. A bit of an edge case, but slightly tilted surfaces have a long history in BHoM.

Plus, minor one: boundary not boundery ;)